src/ploomber_core/telemetry/system_info.py

```diff
@@ -0,0 +1,108 @@
+import os
+from pathlib import Path
+import platform
+import sys
+
+
+def python_version():
+    py_version = sys.version_info
+    return f"{py_version.major}.{py_version.minor}.{py_version.micro}"
+
+
+def is_docker():
+    """Will output if the code is within a container"""
+    try:
+        cgroup = Path("/proc/self/cgroup")
+        docker_env = Path("/.dockerenv")
+        return (
+            docker_env.exists()
+            or cgroup.exists()
+            and any("docker" in line for line in cgroup.read_text().splitlines())
+        )
+    except OSError:
+        return False
+
+
+def get_os():
+    """
+    The function will output the client platform
+    """
+    os = platform.system()
+    if os == "Darwin":
+        return "MacOS"
```

Comment:
is it also possible to get the OS version / Mac M1 or not? Many of the features keep failing on M1. Like some DB docker containers  are not supported

Reply:
this PR only re-organized existing functions, so let's not add new requirements